monai/transforms/croppad/batch.py

```diff
@@ -101,17 +100,22 @@ def __call__(self, batch: Any):
 
                 # If we have a dictionary of data, append to list
                 if is_list_of_dicts:
-                    self.push_transform(batch[idx], key_or_idx, orig_size=orig_size)
+                    self.push_transform(
+                        batch[idx],
+                        key_or_idx,
+                        orig_size=orig_size,
+                        extra_info=self.pop_transform(batch[idx], key_or_idx, check=False),
+                    )
 
         # After padding, use default list collator
         return list_data_collate(batch)
 
     @staticmethod
     def inverse(data: dict) -> Dict[Hashable, np.ndarray]:
-        if not isinstance(data, dict):
+        if not isinstance(data, Mapping):
             raise RuntimeError("Inverse can only currently be applied on dictionaries.")
 
-        d = deepcopy(data)
+        d = dict(data)
```

Comment:
Could you please help search the code base to update all the similar `deepcopy` in the inverse operations?

Thanks in advance.